packages/lisk-p2p/src/peer.ts

```diff
@@ -211,7 +237,11 @@ export class Peer extends EventEmitter {
 			isDiscoveredPeer: true,
 		};
 
-		this._peerDetailedInfo = newPeerInfo;
+		// Some fields from the main _peerInfo cannot be overridden.
+		this._peerDetailedInfo = {
```

Comment:
maybe explicitly assigning what needs to be assign is more clear what cannot be overridden?